### ISSUE TITLE
GIN for regular graph

### ISSUE THREAD

**Issue by Chen-Cai-OSU** [rusty1s/pytorch_geometric#214]
## ❓ Questions & Help

First, thanks a lot for the great framework!

I have a quick question regarding the features of GIN when inputs are regular graphs. I thought in that case, the features on each node of a single graph will always be the same, but when I tried a small example and print out the features are different. 

I was wondering do I miss something here? (I tried to fix all the source of randomness) Why will different nodes of a regular graph have different features?

**Comment by rusty1s** [comment]
Hi, I can not reproduce your issue. I created a test case [here](https://github.com/rusty1s/pytorch_geometric/blob/master/test/nn/conv/test_gin_conv.py):
```python
import torch
from torch.nn import Sequential, Linear, ReLU
from torch_geometric.nn import GINConv

edge_index = torch.tensor([[0, 0, 1, 1, 2, 2, 3, 3, 4, 4, 5, 5],
                               [1, 5, 0, 2, 1, 3, 2, 4, 3, 5, 0, 4]])
    x = torch.ones(6, 1)
    conv = GINConv(Sequential(Linear(1, 8), ReLU(), Linear(8, 8)))

    out = conv(x, edge_index)
    print(out)
```

**Comment by Chen-Cai-OSU** [comment]
Hi Thanks for the quick response. Your example also works for me but when I tried on a larger regular graph there is some problem

    import torch
    from torch.nn import Sequential, Linear, ReLU
    from torch_geometric.nn import GINConv
    edge_index = torch.tensor([[ 1,  1,  1,  1,  1, 21, 21, 21, 21,  7,  7,  7,  7, 18, 18, 18, 18, 18,
         22, 22, 22, 22, 10, 10, 10, 10, 10, 17, 17, 17,  0,  0,  0,  0, 20, 20,
         20, 20, 15, 15, 15, 15, 14, 14, 14, 14,  5,  5,  5,  6, 28, 28, 29, 29,
         11, 11, 24, 24, 26, 26, 26, 26,  3,  3,  3, 19, 19, 19,  9, 16,  4, 13,
         23, 23,  8],
        [21, 11, 24, 23, 12,  7,  6, 13,  5, 16, 29,  2,  8, 22, 23, 16, 17, 29,
          4, 27, 28,  0, 17,  4, 13, 19, 24, 28, 12,  4, 20, 14, 28,  6, 15,  8,
         13,  6,  5, 24,  2, 11,  3,  9, 12,  2,  6, 19, 16, 25, 29, 11,  9, 25,
          3, 27, 26,  8, 25, 16, 27,  4,  9, 12,  8,  2, 13,  9, 23,  2, 25, 27,
         25, 12, 27]])
    # edge_index = torch.tensor([[0, 0, 1, 1, 2, 2, 3, 3, 4, 4, 5, 5],
    #                            [1, 5, 0, 2, 1, 3, 2, 4, 3, 5, 0, 4]])
    # print(np.unique(edge_index.detach().cpu().numpy(), return_counts=True))
    x = torch.ones(30, 1)
    conv = GINConv(Sequential(Linear(1, 8), ReLU(), Linear(8, 8)))
    out = conv(x, edge_index)
    print(out)

**Comment by rusty1s** [comment]
This graph is not regular.

**Comment by Chen-Cai-OSU** [comment]
I forget the direction of edge and that's why I thought the above graph is regular. Thanks a lot. You may close the issue.
